consensus/pow.py

```
@@ -0,0 +1,68 @@
+    Returns:
+        Mined block if successful
+
+    Raises:
+        MiningExceededError if max_nonce is reached
+    """
+
+    target = "0" * difficulty
+    block.nonce = 0
+
+    if logger:
+        logger.info(f"Mining block {block.index} (Difficulty: {difficulty})")
```

Ruff (0.15.0)

[warning] Logging statement uses f-string

(G004)